lib/hydra/derivatives/services/persist_output_file_service.rb

```diff
@@ -4,12 +4,28 @@ class PersistOutputFileService
     # Persists the file within the object at destination_name.  Uses basic containment.
     # If you want to use direct containment (ie. with PCDM) you must use a different service (ie. Hydra::Works::AddFileToGenericFile Service)
     # @param [Object] object the source file is attached to
-    # @param [File] filestream to be added
-    # @param [String] destination_name path to file
-    # @option opts Specific implementations can use this as needed
+    # @param [File] filestream to be added, should respond to :mime_type and :original_name
+    #   original_name will get used as the path for a chile resource in fedora, it is not a path to a file on disk.
 
-    def self.call(object, file, destination_name, opts={})
+    def self.call(object, file)
       raise NotImplementedError, "PersistOutputFileService is an abstract class. Implement `call' on #{self.class.name}"
     end
+
+    def self.determine_original_name( file )
```

Comment:
The [persist_basic_contained_output_file_service_spec](https://github.com/projecthydra/hydra-derivatives/blob/master/spec/services/persist_basic_contained_output_file_service_spec.rb) failed because the method signature was updated, but not the spec. The spec should also cover these 2 new methods.